### PR TITLE
test(mason): update test to be runnable on Windows

### PR DESCRIPTION
**Pull request by mrverdant13**
## Status

**IN DEVELOPMENT**

## Description

- Update tests to be compatible with different line-break and path separation characters.
- Use loopback IP for fake HTTP servers.
- Emulate fake malformed brick by denying file permissions for Windows.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [x] 🧪 Tests

### CODE REVIEW

**Comment by felangel** [comment]
Hi @mrverdant13 👋
Can you provide more context? Tests should already be compatible with windows. The ci runs the tests on Linux, MacOS, and Windows on every pull request.

**Comment by mrverdant13** [comment]
Hi, @felangel!

### Multi-platform CI setup

> The ci runs the tests on Linux, MacOS, and Windows on every pull request.

Actually, only the `mason_cli` package includes a CI config to use Linux, MacOS and Windows. It is the only one that uses the OS matrix.

I could add it if there is no problem with that.

### Windows compatibly

> Tests should already be compatible with windows.

First, I noticed that tests should be run within each package folder thanks to the CI setup, so I did just that.

- Failing tests in the `mason` package

	- Break-line/new-line characters
		Windows uses `\r\n`, while the other platforms use `\n`.
		image

	- Path separation character
		Windows uses `\`, while the other platforms use `/`.
		image

	- Fake malformed brick setup
		While using `chmod` works just fine for Linux and MacOS, the most proper command for Windows is `icacls`.
		image

	- Fake HTTP server setup
		For this kind of purposes, `InternetAddress.loopbackIPv4` is recommended to send all traffic to the same computer.
		The current setup results in an unexpected exception for the tests.
		image

### Subjective context

**OS:** Windows 11
**Dart:** v2.16.1

---

Besides that, I noticed some tests just arbitrarily timeout, but haven't check the root issue with that.

I hope this info is enough!

File: .github/workflows/mason.yaml [hunk 1]
```diff
@@ -36,6 +36,27 @@ jobs:
...
+    defaults:
+      run:
+        working-directory: packages/mason
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
```
Comment (felangel) [inline .github/workflows/mason.yaml:51]:
```suggestion
        os: [macos-latest, windows-2019, ubuntu-latest]
```

**Comment by mrverdant13** [comment]
Alright, this is weird
I am using the same test command that is used in the `mason.yaml`  file, and all tests pass without issues on my local setup.

`dart test -j 1 -x pull-request-only --coverage=coverage`

I believe this is related to Windows VM permissions, but this would be unexpected because GitHub runs all executions as admins by default as described [here](https://docs.github.com/es/actions/using-github-hosted-runners/about-github-hosted-runners). Even user interaction (UAC) is disabled.

Another possibility is that Win command programmatic execution in Dart has limited privileges.

I am testing this in an isolated repo to stop pushing noisy commits and triggering meaningless CI workflows.

**Comment by felangel** [comment]
Closing for now since it's been a while but feel free to reopen whenever you have time to revisit this, thanks!

**Pull request closed without merging**